numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -4166,6 +4166,57 @@ small_correlate(const char * d_, npy_intp dstride,
     }
 }
 
+/*
+*/
+
+/* A clone function for the datetime dtype c_metadata */
+static NpyAuxData *
+_datetime_dtype_metadata_clone(NpyAuxData *data)
+{
+    PyArray_DatetimeDTypeMetaData *newdata =
+        (PyArray_DatetimeDTypeMetaData *)PyArray_malloc(
+                        sizeof(*newdata));
+    if (newdata == NULL) {
+        PyErr_NoMemory();
+        return NULL;
+    }
+
+    memcpy(newdata, data, sizeof(*newdata));
+
+    return (NpyAuxData *)newdata;
+}
+
+/*
+ * Initializes the c_metadata field
+ * must not be static, gcc 4.1.2 on redhat 5 then miscompiles this function
+ * see gh-5163
+ *
+ */
+static NpyAuxData*
```

Comment:
Probably best to leave as `NPY_NO_EXPORT`, given we have no idea if https://github.com/numpy/numpy/issues/5163 is fixed. If you think the issue is now gone, then remove the comment above.

Reply:
removing comment